### PR TITLE
Feature/emoji UI

### PR DESCRIPTION
**Pull request by basheer574**
Pull request

### CODE REVIEW

File: app/src/main/java/com/example/agiledo/ui/adapters/TaskAdapter.kt [hunk 1]
```diff
@@ -18,16 +28,130 @@ class TaskAdapter(var list: List<Task>) : RecyclerView.Adapter<TaskAdapter.TaskH
...
+                })
+
+                emoji_ten.setOnClickListener(View.OnClickListener {
+                    selected.setText(emoji_ten.text)
+                })
+
+                dialog.show()
+                okBtn.setOnClickListener{
+                    //TODO: Store the new emoji code in sql database.
+                    val x = selected.text
```
[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/java/com/example/agiledo/ui/adapters/TaskAdapter.kt:143]: done , the name was changed to selectedEmoji

File: app/src/main/res/layout/emoji_dialog.xml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_margin="@dimen/marginNormal"
+    android:elevation="8dp"
+    >
+
+
+    <com.aghajari.emojiview.view.AXEmojiEditText
+        android:id="@+id/selected_emoji"
+        android:layout_width="32dp"
```
[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/res/layout/emoji_dialog.xml:14]: done, replaced with paddingGaint

File: app/src/main/res/layout/emoji_dialog.xml [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        android:layout_height="wrap_content"
+        android:layout_marginStart="100dp"
+        android:layout_marginTop="56dp"
+        android:text="1"
+        android:textSize="20sp"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <com.aghajari.emojiview.view.AXEmojiTextView
+        android:id="@+id/emoji2"
```
[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/res/layout/emoji_dialog.xml:38]: Done

[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/res/layout/emoji_dialog.xml:38]: And thanks for the tip ;)

File: app/src/main/java/com/example/agiledo/ui/adapters/TaskAdapter.kt [hunk 4]
```diff
@@ -18,16 +28,130 @@ class TaskAdapter(var list: List<Task>) : RecyclerView.Adapter<TaskAdapter.TaskH
...
     override fun onBindViewHolder(holder: TaskHolder, position: Int) {
+
         var currentItem = list[position]
         holder.binding.apply {
             taskTitle.text = currentItem.taskName
             taskDescription.text = currentItem.taskDescription
             taskStartDate.text = currentItem.taskStartDate
             taskDueDate.text = currentItem.taskDueDate
             taskAssignedTo.text= currentItem.taskAssignedTo
+            personFirstLetter.text = currentItem.taskAssignedTo.first().toString()
```
[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/java/com/example/agiledo/ui/adapters/TaskAdapter.kt:39]: Done , but i Still need to take the first letter of the current user, so this line of code is unchanged.

File: app/src/main/java/com/example/agiledo/ui/HomeFragment.kt [hunk 5]
```diff
@@ -17,6 +18,5 @@ class HomeFragment : BaseFragment<FragmentHomeBinding>() {
     }
 
     override fun addCallbacks() {
```
[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/java/com/example/agiledo/ui/HomeFragment.kt:20]: I don't remember removing theme, but i will add theme.

[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/java/com/example/agiledo/ui/HomeFragment.kt:20]: Done

File: app/src/main/res/drawable/emoji_edittext_shape.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+<item>
+    <shape>
+        <gradient android:startColor="#CFD8DC" android:endColor="#B0BEC5"></gradient>
```
[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/res/drawable/emoji_edittext_shape.xml:5]: Done

File: app/src/main/res/layout/item_task.xml [hunk 7]
```diff
@@ -16,14 +17,19 @@
...
-        <ImageView
-            android:id="@+id/pencil_image"
-            android:layout_width="24dp"
-            android:layout_height="24dp"
-            android:rotationY="180"
-            android:background="@drawable/ic_pencil"
+
+        <com.aghajari.emojiview.view.AXEmojiEditText
+            android:id="@+id/task_image"
+            android:layout_width="40dp"
```
[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/res/layout/item_task.xml:23]: Done

File: app/src/main/res/drawable/emoji_edittext_shape.xml [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+        <corners android:radius="20dp"></corners>
```
[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/res/drawable/emoji_edittext_shape.xml:6]: Done

File: app/src/main/java/com/example/agiledo/ui/adapters/TaskAdapter.kt [hunk 9]
```diff
@@ -18,16 +28,130 @@ class TaskAdapter(var list: List<Task>) : RecyclerView.Adapter<TaskAdapter.TaskH
...
+                emoji_five.text = AXEmojiUtils.replaceWithEmojis(holder.binding.root.context,emojiUnicode_five,20F)
+                emoji_six.text = AXEmojiUtils.replaceWithEmojis(holder.binding.root.context,emojiUnicode_six,20F)
+                emoji_seven.text = AXEmojiUtils.replaceWithEmojis(holder.binding.root.context,emojiUnicode_seven,20F)
+                emoji_eight.text = AXEmojiUtils.replaceWithEmojis(holder.binding.root.context,emojiUnicode_eight,20F)
+                emoji_nine.text = AXEmojiUtils.replaceWithEmojis(holder.binding.root.context,emojiUnicode_nine,20F)
+                emoji_ten.text = AXEmojiUtils.replaceWithEmojis(holder.binding.root.context,emojiUnicode_ten,20F)
+
+
+                selected.inputType = InputType.TYPE_NULL
+                cancelBtn.setOnClickListener{
```
[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/java/com/example/agiledo/ui/adapters/TaskAdapter.kt:91]: Done , function added

File: app/src/main/java/com/example/agiledo/ui/adapters/TaskAdapter.kt [hunk 10]
```diff
@@ -18,16 +28,130 @@ class TaskAdapter(var list: List<Task>) : RecyclerView.Adapter<TaskAdapter.TaskH
...
+                val emojiUnicode_four = AXEmojiUtils.getEmojiUnicode(0x1F4AC)
+                val emojiUnicode_five = AXEmojiUtils.getEmojiUnicode(0x1F4BF)
+                val emojiUnicode_six = AXEmojiUtils.getEmojiUnicode(0x1F4C3)
+                val emojiUnicode_seven = AXEmojiUtils.getEmojiUnicode(0x1F4CC)
+                val emojiUnicode_eight = AXEmojiUtils.getEmojiUnicode(0x1F4CE)
+                val emojiUnicode_nine = AXEmojiUtils.getEmojiUnicode(0x1F4D7)
+                val emojiUnicode_ten = AXEmojiUtils.getEmojiUnicode(0x1F4DE)
+
+
+                emoji_one.text = AXEmojiUtils.replaceWithEmojis(holder.binding.root.context,emojiUnicode_one,20F)
```
[the remark this answers is not in the document]
Comment (basheer574) [reply app/src/main/java/com/example/agiledo/ui/adapters/TaskAdapter.kt:78]: Done, function created not exactly as you said but it worked

**Comment by wesamnabeel99** [comment]
great work, I'll finally merge it

**Comment by wesamnabeel99** [comment]
great work, I'll finally merge it

**Comment by wesamnabeel99** [comment]
great work, I'll finally merge it

**Pull request merged**